src/main/java/com/shopify/ShopifySdk.java

```diff
@@ -489,6 +620,16 @@ public ShopifyOrder createOrder(final ShopifyOrderCreationRequest shopifyOrderCr
 		return shopifyOrderRootResponse.getOrder();
 	}
 
+	public ShopifyOrder updateOrderShippingAddress(
```

Comment:
I like the direction you are going with a builder here that takes in a bunch of optional steps but I think you need to take it one step further so that this is just a generic `updateOrder` method.

I would like the signature to look like this:

```
public ShopifyOrder updateOrder(
			final ShopifyOrderUpdateRequest shopifyOrderUpdateRequest);
```

Reply:
@rjdavis3 this is the problem we talked about a couple of weeks ago. jackson doesn't support only marshalling certain null fields. My vote would be to stay away from having a generic update order method as we won't be able to support nulling out fields or risk getting validation by providing nulls for everything not supplied. (unless we do a GET)

Reply:
@ryankazokas ok I get that point. In that case I think a signature like this is fine but we should change `ShopifyOrderShippingAddressUpdateRequest` to have a step builder pattern which build outs out `ShopifyAddressUpdateRequest`. Then we can hide `ShopifyAddressUpdateRequest` from the user.

The builder for `ShopifyOrderShippingAddressUpdateRequest` will look like this:

```
ShopifyOrderShippingAddressUpdateRequest	.newBuilder().withId(someShopifyOrderId).withAddress1(address1).withAddress2(address2)
...
.build();
```

The only thing I am unsure of is whether all of the shipping address steps (i.e. withAddress1, withAddress2) should be optional steps. I'm thinking they should not be optional steps because otherwise they might not know they are nulling out fields they did not provide.

Reply:
Sounds good @rjdavis3 . Makes sense. I'll make all of the fields required in the step builder.